### DISCUSSION TITLE
linguist overrides in .gitattributes are all ignored

### DISCUSSION THREAD

**Discussion by phorward** [github-linguist/linguist#5857] [Classification] [answered]
Hello!

I have a web-app written mostly in Python. Anyway, github-linguist classifies most of its source as "JavaScript", as the repository contains lots of vendored content. Therefore I decided to override it in the `.gitattributes` file like this:
```
$ cat .gitattributes 
vi/public/htmleditor/htmleditor.min.js linguist-generated
vi/public/js/vi.min.js linguist-generated
vi/public/scripter/**/*.js linguist-vendored
```
And testing if the attributes matches is the case, like here:
```
$ git check-attr --all vi/public/scripter/codemirror-5.61.0/src/edit/main.js 
vi/public/scripter/codemirror-5.61.0/src/edit/main.js: text: set
vi/public/scripter/codemirror-5.61.0/src/edit/main.js: linguist-vendored: set
vi/public/scripter/codemirror-5.61.0/src/edit/main.js: eol: lf
```
Now, when I run github-linguist, the result is always and forever
```
$ ~/.gem/ruby/3.0.0/bin/github-linguist 
69.09%  4214354    JavaScript
18.17%  1108470    HTML
6.23%   379830     Python
5.23%   318896     CSS
1.27%   77716      Less
0.01%   562        Makefile
0.01%   352        Shell
```

Even when I change the `.gitattributes` to handle all `.js`-files as C source code, the result remains the same...
```
$ cat .gitattributes 
*.js linguist-language=C
$ git check-attr --all vi/public/scripter/codemirror-5.61.0/src/edit/main.js 
vi/public/scripter/codemirror-5.61.0/src/edit/main.js: text: set
vi/public/scripter/codemirror-5.61.0/src/edit/main.js: linguist-language: C
vi/public/scripter/codemirror-5.61.0/src/edit/main.js: eol: lf
$ ~/.gem/ruby/3.0.0/bin/github-linguist 
69.09%  4214354    JavaScript
18.17%  1108470    HTML
6.23%   379830     Python
5.23%   318896     CSS
1.27%   77716      Less
0.01%   562        Makefile
0.01%   352        Shell
$ ~/.gem/ruby/3.0.0/bin/github-linguist --version
github-linguist 7.20.0
```

What am I doing wrong???

**Comment by lildude** [comment, marked as answer]
Sounds like you might have missed this from the [override docs](https://github.com/github/linguist/blob/master/docs/overrides.md):

> When testing with a local installation of Linguist, **take note that the added attributes will _not_ take effect until the `.gitattributes` file is committed to your repository.**

**Reply by phorward** [reply]
> Sounds like you might have missed this from the [override docs](https://github.com/github/linguist/blob/master/docs/overrides.md):
> 
> > When testing with a local installation of Linguist, **take note that the added attributes will _not_ take effect until the `.gitattributes` file is committed to your repository.**

Ah! Okay that was the mistake. Thanks!
